src/main/telemetry/crsf.c

```diff
@@ -266,19 +266,35 @@ void crsfFrameFlightMode(sbuf_t *dst)
     uint8_t *lengthPtr = sbufPtr(dst);
     sbufWriteU8(dst, 0);
     sbufWriteU8(dst, CRSF_FRAMETYPE_FLIGHT_MODE);
-
-    // use same logic as OSD, so telemetry displays same flight text as OSD
-    const char *flightMode = "ACRO";
-    if (airmodeIsEnabled()) {
-        flightMode = "AIR";
-    }
+    const char *flightMode = "";
+    
     if (FLIGHT_MODE(FAILSAFE_MODE)) {
-        flightMode = "!FS";
-    } else if (FLIGHT_MODE(ANGLE_MODE)) {
-        flightMode = "STAB";
-    } else if (FLIGHT_MODE(HORIZON_MODE)) {
-        flightMode = "HOR";
+        flightMode = "!FS!";
+    } else if (isArmingDisabled()) {
+        flightMode = "!ERR";
+    } else if (ARMING_FLAG(ARMED)) {
+        // Armed flight modes in decreasing order of importance
+        } if (FLIGHT_MODE(GPS_RESCUE_MODE)) {
+            flightMode = "RTH";
+        } else if (FLIGHT_MODE(PASSTHRU_MODE)) {
+            flightMode = "MANU";
+        } else if (FLIGHT_MODE(ANGLE_MODE)) {
+            flightMode = "STAB";
+        } else if (FLIGHT_MODE(HORIZON_MODE)) {
+            flightMode = "HOR";
+        } else if (airmodeIsEnabled()) {
+            flightMode = "AIR";
+        } else {
+            flightMode = "ACRO";
+        }
+#ifdef USE_GPS
+    } else if (featureIsEnabled(FEATURE_GPS) && (!STATE(GPS_FIX) || !STATE(GPS_FIX_HOME))) {
+        flightMode = "WAIT"; // Waiting for GPS lock
+#endif
+    } else {
+        flightMode = "OK"; // Ready to arm
```

Comment:
Please revert this and stop sending meaningless information.

Reply:
This does not make sense at all in Betaflight.

A craft is always ready to arm, unless `isArmingDisabled()` is true. So the pilot can assume that unless they get `!ERR` the craft is ready to arm. There is no point in withholding safety relevant flight mode information just to send a meaningless status.